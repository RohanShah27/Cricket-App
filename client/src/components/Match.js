import React, { Component, useState, useEffect } from "react";
import { connect } from "react-redux";
import "../styles/match.css";
import india from "../assests/india.jpg";
import pakistan from "../assests/pakistan.jpg";
import { getMatchesByType } from "../actions/matchActions";
var json;
export class Match extends Component {
  constructor(props) {
    super(props);
    // this.getData = this.getData.bind(this);
  }

  componentDidMount() {
    let type = { type: "ODI" };
    this.props.getMatchesByType(type);
  }

  state = {
    type: "",
    matchType: "ODI",
    activeClass: "match-active-option",
    data: [],
    userId: 1,
    loading: false
  };
  sendType() {
    console.log("type" + this.state.type);
    let matches = {
      type: this.state.type
    };
    this.props.getMatchesByType(matches);
  }

  sendData(types) {
    this.setState({ matchType: types });
    console.log(types);
    this.setState({
      type: types
    });
    this.sendType();
  }

  render() {
    console.log(this.props.match.result);
    return (
      <div>
        <div className="match-pc-tab">
          <input
            defaultChecked="defaultChecked"
            id="match-tab1"
            type="radio"
            name="pct"
            onClick={() => this.sendData("ODI")}
          />
          <input
            id="match-tab2"
            type="radio"
            name="pct"
            onClick={() => this.sendData("Test")}
          />
          <input
            id="match-tab3"
            type="radio"
            name="pct"
            onClick={() => this.sendData("T20")}
          />
          <nav>
            <ul>
              <li className="match-tab1">
                <label htmlFor="match-tab1">
                  <b>ODI</b>
                </label>
              </li>
              <li className="match-tab2">
                <label htmlFor="match-tab2">
                  <b>TEST</b>
                </label>
              </li>
              <li className="match-tab3">
                <label htmlFor="match-tab3">
                  <b>T20</b>
                </label>
              </li>
            </ul>
          </nav>
          <section>
            <div className="match-tab1">
              <div className="match-testimonials">
                {this.props.viewmatch.map(match => (
                  <div
                    className="match-card"
                    onClick={() => {
                      this.props.history.push(
                        "/matchdetails/" + match.match_id,
                        { match }
                      );
                    }}
                  >
                    <div className="match-parent">
                      <div className="match-first">
                        <img className="match-img" src={india} />
                        <p>{match.team1}</p>
                        <p className="match-p">250/5 (50.0)</p>
                      </div>
                      <div className="match-second">
                        <img className="match-img" src={pakistan} />

                        <p>{match.team2}</p>
                        <p className="match-p">224/6 (50.0)</p>
                      </div>

                      <div className="match-third">
                        {match.match_winner} {match.won_by}
                        <p>MoM: {match.player_of_the_match}</p>
                      </div>
                    </div>
                  </div>
                ))}
              </div>
            </div>
          </section>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  viewmatch: state.matchReducers.viewmatch
});

export default connect(
  mapStateToProps,
  { getMatchesByType }
)(Match);
