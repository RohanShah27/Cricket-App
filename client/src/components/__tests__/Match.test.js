import React from "react";
import { shallow, mount } from "enzyme";
import { Match } from "../Match";
const history = { push: jest.fn() };

let matches = [
  {
    match_id: 2,
    match_competition: "Big bAsh leagues",
    match_type: "ODI",
    team1: "India",
    team2: "Pakistan",
    team1_image: "in",
    team2_image: "pk"
  },
  {
    match_id: 3,
    match_competition: "Big bAsh leagues",
    match_type: "T20",
    team1: "India",
    team2: "Srilanka",
    team1_image: "in",
    team2_image: "lk"
  }
];

const match = jest.fn();
const getMatchesByType = jest.fn();
const wrapper = mount(
  <Match match={match} getMatchesByType={getMatchesByType} matches={matches} />
);

describe("Test Match Component", () => {
  it("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should check for component did mount as a function", () => {
    const componentDidMount = jest.spyOn(Match.prototype, "componentDidMount");
    wrapper.instance().componentDidMount();
    expect(componentDidMount).toHaveBeenCalled();
  });
  it("should check if Infinite Scroll component is present or not", () => {
    expect(wrapper.find("InfiniteScroll").exists()).toBeTruthy();
  });

  it("should have a conatiner with name matchtype-tab", () => {
    expect(wrapper.find("#matchtypetab")).toBeTruthy();
  });

  it("should have a conatiner with name matchCard", () => {
    expect(wrapper.find("#matchCard")).toBeTruthy();
  });
  it("should have a match team one in matches ", () => {
    expect(wrapper.find("#matchTeamOne0").props().children).toBe(
      matches[0].team1
    );
  });
  it("should have a match team two in matches ", () => {
    expect(wrapper.find("#matchTeamTwo0").props().children).toBe(
      matches[0].team2
    );
  });

  it("should have a match team one in matches ", () => {
    expect(wrapper.find("#matchTeamOne1").props().children).toBe(
      matches[1].team1
    );
  });
  it("should have a match team two in matches ", () => {
    expect(wrapper.find("#matchTeamTwo1").props().children).toBe(
      matches[1].team2
    );
  });
});
