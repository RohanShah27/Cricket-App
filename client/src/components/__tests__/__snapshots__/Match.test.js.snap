// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Match Component should render the component 1`] = `
<Match
  getMatchesByType={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "gender": undefined,
            "type": "ODI",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  match={[MockFunction]}
  matches={
    Array [
      Object {
        "match_competition": "Big bAsh leagues",
        "match_id": 2,
        "match_type": "ODI",
        "team1": "India",
        "team1_image": "in",
        "team2": "Pakistan",
        "team2_image": "pk",
      },
      Object {
        "match_competition": "Big bAsh leagues",
        "match_id": 3,
        "match_type": "T20",
        "team1": "India",
        "team1_image": "in",
        "team2": "Srilanka",
        "team2_image": "lk",
      },
    ]
  }
>
  <div>
    <div
      className="match-pc-tab"
      id="matchtype-tab"
    >
      <input
        defaultChecked="defaultChecked"
        id="match-tab1"
        name="pct"
        onClick={[Function]}
        type="radio"
      />
      <input
        id="match-tab2"
        name="pct"
        onClick={[Function]}
        type="radio"
      />
      <input
        id="match-tab3"
        name="pct"
        onClick={[Function]}
        type="radio"
      />
      <nav>
        <ul>
          <li
            className="match-tab1"
          >
            <label
              htmlFor="match-tab1"
            >
              <b>
                ODI
              </b>
            </label>
          </li>
          <li
            className="match-tab3"
          >
            <label
              htmlFor="match-tab3"
            >
              <b>
                T20
              </b>
            </label>
          </li>
        </ul>
      </nav>
      <section>
        <div
          className="match-tab1"
        >
          <InfiniteScroll
            dataLength={0}
            disableBrowserPullToRefresh={true}
            endMessage={
              <p
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <b>
                  Yay! You have seen all Teams
                </b>
              </p>
            }
            hasMore={true}
            height={600}
            loader={
              <div
                className="loader-container"
              >
                <div
                  className="user-loader"
                />
              </div>
            }
            next={[Function]}
            pullDownToRefreshContent={
              <h3>
                Pull down to refresh
              </h3>
            }
            pullDownToRefreshThreshold={100}
            releaseToRefreshContent={
              <h3>
                Release to refresh
              </h3>
            }
          >
            <div
              style={Object {}}
            >
              <div
                className="infinite-scroll-component "
                style={
                  Object {
                    "WebkitOverflowScrolling": "touch",
                    "height": 600,
                    "overflow": "auto",
                  }
                }
              >
                <div
                  className="match-testimonials"
                />
                <div
                  className="loader-container"
                >
                  <div
                    className="user-loader"
                  />
                </div>
              </div>
            </div>
          </InfiniteScroll>
        </div>
      </section>
    </div>
  </div>
</Match>
`;
